[docs] Update limitations for YSQL Connection Manager (#25751)
yugabyte/yugabyte-db: docs/content/stable/explore/going-beyond-sql/connection-mgr-ysql.md

- Currently, you can't apply custom configurations to individual pools. The YSQL Connection Manager configuration applies to all pools.
- When YSQL Connection Manager is enabled, the backend PID stored using JDBC drivers may not be accurate. This does not affect backend-specific functionalities (for example, cancel queries), but this PID should not be used to identify the backend process.
- By default, `currval` and `nextval` functions do not work when YSQL Connection Manager is enabled. They can be supported with the help of the `ysql_conn_mgr_sequence_support_mode` flag.

> - Queries using the extended query protocol may exhibit unexpected behavior with certain drivers (e.g., libpq (C), Npgsql (C#), lib/pq (Go)), though extensive testing shows compatibility with the JDBC and psycopg/psycopg2 drivers.[#24898](https://github.com/yugabyte/yugabyte-db/issues/24898)

Reviewer: ```suggestion
- Queries using the extended query protocol may exhibit unexpected behavior with certain drivers (including libpq (C), Npgsql (C#), lib/pq (Go)), though extensive testing shows compatibility with the JDBC and psycopg/psycopg2 drivers. [#24898](https://github.com/yugabyte/yugabyte-db/issues/24898)
```

- YSQL Connection Manager does not yet support IPv6 connections. [#24765](https://github.com/yugabyte/yugabyte-db/issues/24765)

> - Certificate-based TLS authentication is currently not supported with YSQL Connection Manager [#20658](https://github.com/yugabyte/yugabyte-db/issues/20658)

Reviewer: ```suggestion
- YSQL Connection Manager does not currently support certificate-based TLS authentication. [#20658](https://github.com/yugabyte/yugabyte-db/issues/20658)
```

> - Certificate-based TLS authentication is currently not supported with YSQL Connection Manager [#20658](https://github.com/yugabyte/yugabyte-db/issues/20658)

Reviewer: "Certificate-based" - is there another kind? 
But it does support TLS encryption?
May need to be clearer here

Author: afaik encryption can be enabled even without the need for certificates (sslmode=require, for example)

we can change it to be "TLS encryption" instead of "TLS authentication" for clarity

> - YSQL Connection Manager does not currently support certificate-based TLS authentication. [#20658](https://github.com/yugabyte/yugabyte-db/issues/20658)

Author: ```suggestion
- YSQL Connection Manager does not currently support certificate-based TLS encryption. [#20658](https://github.com/yugabyte/yugabyte-db/issues/20658)
```

- Currently, you can't apply custom configurations to individual pools. The YSQL Connection Manager configuration applies to all pools.
- When YSQL Connection Manager is enabled, the backend PID stored using JDBC drivers may not be accurate. This does not affect backend-specific functionalities (for example, cancel queries), but this PID should not be used to identify the backend process.
- By default, `currval` and `nextval` functions do not work when YSQL Connection Manager is enabled. They can be supported with the help of the `ysql_conn_mgr_sequence_support_mode` flag.
- Queries using the extended query protocol may exhibit unexpected behavior with certain drivers (including libpq (C), Npgsql (C#), lib/pq (Go)), though extensive testing shows compatibility with the JDBC and psycopg/psycopg2 drivers. [#24898](https://github.com/yugabyte/yugabyte-db/issues/24898)
- YSQL Connection Manager does not yet support IPv6 connections. [#24765](https://github.com/yugabyte/yugabyte-db/issues/24765)

> - YSQL Connection Manager does not currently support certificate-based TLS encryption. [#20658](https://github.com/yugabyte/yugabyte-db/issues/20658)

Author: ```suggestion
- Currently, [auth-method](https://docs.yugabyte.com/preview/secure/authentication/host-based-authentication/#auth-method) `cert` is not supported for host-based authentication. [#20658](https://github.com/yugabyte/yugabyte-db/issues/20658)
```

> - Using auth-backends (`ysql_conn_mgr_use_auth_backend=true`) for the authentication of logical connections results in higher connection acquisition latencies. It is still recommended to consult the YSQL Connection Manager Dev team before using the auth-passthrough (`ysql_conn_mgr_use_auth_backend=false`) implementation. [#25313](https://github.com/yugabyte/yugabyte-db/issues/25313)

Reviewer: ```suggestion
- Although the use of auth-backends (`ysql_conn_mgr_use_auth_backend=true`) to authenticate logical connections can result in higher connection acquisition latencies, using auth-passthrough (`ysql_conn_mgr_use_auth_backend=false`) may not be suitable depending on your workload. Contact the YSQL Connection Manager Development team before setting ysql_conn_mgr_use_auth_backend to false. [#25313](https://github.com/yugabyte/yugabyte-db/issues/25313)
```
